src/core/core.js

```diff
@@ -280,7 +281,7 @@ export default class Core {
         this._persistLocationRadius();
         this._reportFollowUpQueryEvent(data[StorageKeys.QUERY_ID], Searcher.VERTICAL);
 
-        this.storage.set(StorageKeys.QUERY_ID, data[StorageKeys.QUERY_ID]);
+        this.storage.setWithPersist(StorageKeys.QUERY_ID, data[StorageKeys.QUERY_ID]);
```

Comment:
Did product consider any potential issues with persisting the query id?

Reply:
Nope, Davish seemed cool with it: https://yext.slack.com/archives/C016ZKY42CF/p1651009932053909?thread_ts=1651008679.567229&cid=C016ZKY42CF

Reply:
I was thinking along the lines of somebody bookmarking the page, copy pasting the URL to send to somebody else, or copy pasting the url to be linked to from a different page.

Reply:
Oh I wrote my above comment before seeing Tom's (github race condition), sgtm!

Reply:
I was thinking in the case of someone perform a search and open the vertical page in a new tab, which would still consider valid to fire a follow_up_query. This distinction between opening a vertical page in a new tab vs copying the link to a new tab can be resolve with checking document.referrer. My concern would be on page refresh, I will look into this further.

Reply:
I added logic to guard the case you and tom mentioned, by ensuring that the navigation is due to a refresh or from a tab href link. For refresh, I did use an experimental [PerformanceNavigationTiming](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming/type) API which was recommended as oppose to the deprecated [PerformanceNavigation API](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigation/type).